src/components/FilterPanel/FilterPanel.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { describe, it, expect, vi } from 'vitest';
+
+import { FilterPanel } from './FilterPanel';
+
+describe('FilterPanel', () => {
+  it('resalta el filtro seleccionado', () => {
+    render(
+      <FilterPanel
+        filters={[
+          { label: 'All Tests', value: 'all' },
+          { label: 'Upcoming', value: 'upcoming' },
+        ]}
+        selectedValue="all"
+        onSelect={() => {}}
+      />
+    );
+
+    const selectedButton = screen.getByTestId('filter-all');
+    expect(selectedButton).toHaveAttribute('aria-checked', 'true');
+  });
+
+  it('dispara onSelect al hacer clic en un filtro', () => {
+    const mockSelect = vi.fn();
+
+    render(
+      <FilterPanel
+        filters={[
+          { label: 'All Tests', value: 'all' },
+          { label: 'Upcoming', value: 'upcoming' },
+        ]}
+        selectedValue="all"
+        onSelect={mockSelect}
+      />
+    );
+
+    const upcomingButton = screen.getByTestId('filter-upcoming');
```

Comment:
This test queries the upcoming button by data-testid and uses fireEvent.click. Prefer accessible queries and user-event for more realistic interactions.

**Sugerencia:**
```typescript
Replace with:

const upcomingButton = screen.getByRole('radio', { name: /Upcoming/i });
await userEvent.click(upcomingButton);

(Install @testing-library/user-event as dev dependency if not present.)
```